runners/spark/spark_runner.gradle

```diff
@@ -166,6 +166,10 @@ dependencies {
   provided "org.apache.spark:spark-network-common_$spark_scala_version:$spark_version"
   provided "org.apache.spark:spark-sql_$spark_scala_version:$spark_version"
   provided "org.apache.spark:spark-streaming_$spark_scala_version:$spark_version"
+  runtimeOnly "org.apache.spark:spark-core_$spark_scala_version:$spark_version"
```

Comment:
Note that `provided` is used when generating the pom.xml for this artifact to list a provided dependency.

Reply:
> 1. Why do we need to have two configurations (`provided` and `runtimeOnly`) for the same deps?
> 2. Can you make sure that generated jars won't cause breaking changes for users who already use any of Spark Runners?

1. because provided scope is given by a gradle plugin and is not equivalent to maven provided scope, it lacks putting the libs in the runtime classpath. They are used for local in memory runs for example Nexmark.
2. I Like I said in the description of the PR I checked that the generated jars do not contain apache spark libs.

Reply:
> Note that `provided` is used when generating the pom.xml for this artifact to list a provided dependency.

I also took a look at the generated poms (thanks Alexey for the ping about that) and they are exactly identical to the ones before the change for both spark 2 and spark 3 modules. Meaning that they contain apache spark libs in provided scope as before.